api/urls.py

```diff
@@ -98,7 +98,16 @@ def register_url(self, subclass, viewset):
     url(r'atom/?', views.CreativeWorksAtom(), name='atom'),
     url(r'graph/?', GraphQLView.as_view(graphiql=True)),
     url(r'userinfo/?', ensure_csrf_cookie(views.ShareUserView.as_view()), name='userinfo'),
-    url(r'search/(?!.*_bulk\/?$)(?P<url_bits>.*)', csrf_exempt(views.ElasticSearchView.as_view()), name='search'),
+
+    # only match _count, _search, _suggest requests
+    url(
+        r'search/(?!.*_bulk\/?$)(?P<url_bits>(_|[a-zA-Z]+/_)((?=search)|(?=count)|(?=suggest))[a-zA-Z]{5,7}/{0,1}$)',
```

Comment:
Could you simplify this regex a bit?

if you don't need to I would recommend against positive look aheads, they're [a bit odd](http://www.regular-expressions.info/lookaround.html).
This would be much more clear as `(?:_search|_count|_suggest)`

I think you can remove the `_bulk` now that this is a whitelist endpoint.

If it is easier to understand and/or other easier to handle, it might make sense to separate out endpoints that we allow POSTing, GETTing, or POST and GET?
Just an off hand idea.